backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -232,4 +235,12 @@ public int getDedicatedCount(Guid vdsId) {
         return (int) resourceManager.getVdsManager(vdsId).getCpuTopology().stream()
                 .filter(VdsCpuUnit::isDedicated).count();
     }
+
+    private Set<Integer> getOnlineSockets(List<VdsCpuUnit> cpuTopology) {
```

Comment:
why returning a set here and not a list?

Reply:
You can't get duplicates here. List will do, but Set is better in my opinion.

Reply:
there is no benefit in having a set if you don't intend to search for particular element(s) and don't filter duplicates - it just returns a more complicated data structure.. how about returning the stream instead if you just iterate it?

Reply:
then i can't do forEach within the for loop, will List work for you?

Reply:
yes, it would have required to change the for-loop to forEach
returning a list is also ok